packages/astro/src/core/build/plugins/util.ts

```diff
@@ -75,13 +75,14 @@ export function shouldInlineAsset(
 	assetPath: string,
 	assetsInlineLimit: NonNullable<BuildOptions['assetsInlineLimit']>
 ) {
-	if (typeof assetsInlineLimit === 'number') {
-		return Buffer.byteLength(assetContent) < assetsInlineLimit;
+	if (typeof assetsInlineLimit === 'function') {
+		const result = assetsInlineLimit(assetPath, Buffer.from(assetContent));
+		if (result != null) {
+			return result;
+		}
 	}
-
-	const result = assetsInlineLimit(assetPath, Buffer.from(assetContent));
-	if (result != null) {
-		return result;
+	if (typeof assetsInlineLimit === 'number' || typeof assetsInlineLimit === 'string') {
```

Comment:
I think we can change this to
```suggestion
	} else {
```

Since if it's not a function, it should already be either a number or string

Reply:
Yeah I think we should change it.

Reply:
If I simply change this to an `else`, the `4096` default fallback at the bottom of the function will never fire. I would have to at least check `else if (typeof assetsInlineLimit !== 'undefined')`. It might not be set at all.

Would you like me to do that instead of the string/number check?

Reply:
It will fire if `assetsInlineLimit` is a function but the function returns undefined. `assetsInlineLimit` should already be set by default because the param has `NonNullable`.

Maybe to make it less confusing, we can move the `4096` fallback inside the `typeof assetsInlineLimit === 'function'` block so it's less confusing.

Reply:
We currently type it as `assetsInlineLimit: NonNullable<BuildOptions['assetsInlineLimit']>`. That means it can only be either a string or a function. It can't be nullable because we marked it as `NonNullable` (the consumers will always set a value). In which case, we don't need to cover the 3rd case. We should able to trust the types here.

Overall, my suggestion would be this kind of flow:

```js
	if (typeof assetsInlineLimit === 'function') {
		const result = assetsInlineLimit(assetPath, Buffer.from(assetContent));
		if (result != null) {
			return result;
		} else {
			return Buffer.byteLength(assetContent) < 4096; // Fallback to 4096kb by default (same as Vite)
		}
	}

	return Buffer.byteLength(assetContent) < Number(assetsInlineLimit);
```

Reply:
Sure. I'd consider my solution safer (not everyone uses TypeScript), but I understand your want to simplify, so I've updated that now.